src/rs_wall_clock.h

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2006-Present, Redis Ltd.
+ * All rights reserved.
+ *
+ * Licensed under your choice of the Redis Source Available License 2.0
+ * (RSALv2); or (b) the Server Side Public License v1 (SSPLv1); or (c) the
+ * GNU Affero General Public License v3 (AGPLv3).
+*/
+
+#pragma once
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <time.h>
+#include <stdint.h>
+
+typedef struct {
+    struct timespec start;
+} rs_wall_clock;
```

Comment:
Consider renaming `start` to `time` or a more generic name (`end->start` below is confusing), or simplify the definition, since there are no other attributes in the struct:
```suggestion
typedef struct timespec rs_wall_clock;
```